File: frontend__v2/subdealer_admin_panel/src/helpers/BackendAuthHelper.js
import axios from "axios"
import { BASE_URL } from "../Config"

export const loginAtBackend = async ({ phone, password, onSuccess, onError }) => {
    const formData = new FormData();
    formData.append('phone', phone)
    formData.append('password', password)

    await axios({
        method: 'POST',
        url: `${BASE_URL}users/subdealers/login/`,
        data: formData
    }).then(
        (e) => {
            localStorage.setItem('jwt', e.data.jwt);
            localStorage.setItem('phone', e.data.user.phone)
            onSuccess(e)
        }
    ).catch(
        (err) => {
            onError({ err });
        }
    )
}

export const checkAuthSession = async () => {
    if (!(localStorage.getItem('jwt') && localStorage.getItem('phone'))) {
        return false;
    }

    let res = null;

    await axios({
        url: `${BASE_URL}users/check_session/${localStorage.getItem('phone')}/`,
        method: 'get',
        headers: { 'Authorization': localStorage.getItem('jwt') },
    }).then(e => {
        if (e.status === 200) {
            res = e.data.exists
        }
    })
    .catch((err) => {
        res = false;
    })

    return res;

}

export const logoutAtBackend = async ({ onSuccess, onError }) => {
    await axios({
        url: `${BASE_URL}users/logout/`,
        method: 'GET',
        headers: { 'Authorization': localStorage.getItem('jwt') },
    }).then(e => {
        onSuccess()
    }).catch(err => {
        onError({ err });
    })
}

export const createSubdealerAccAtBackend = async ({regFormData, onSuccess, onError}) => {
    const formData = new FormData();
    formData.append('is_subdealer', true)
    if (regFormData.password !== regFormData.confPassword) {
        return;
    }

    Object.keys(regFormData).forEach((val, attribute)=>{
        if (val !== 'confPassword') {
            if(regFormData[val] !== '') {
                formData.append(val, regFormData[val]);
            }
        }
    });

    await axios({
        url : `${BASE_URL}users/subdealers/`,
        method : 'POST',
        data : formData
    }).then(e=>{
        localStorage.setItem('phone', e.data.user.phone)
        localStorage.setItem('jwt', e.data.jwt)
        onSuccess(e.data)
    }).catch(err=>{
        onError({err})
    })

}

File: frontend__v2/shoppy/lib/helpers/BackendAuthHelper.dart
import 'dart:convert';
import 'dart:math';

import 'package:shared_preferences/shared_preferences.dart';
import 'package:shoppy/Env.dart';
// ignore: import_of_legacy_library_into_null_safe
import 'package:http/http.dart' as http;

class BackendAuthHelper {
  final BASE_URL = Env().BASE_URL;
  late SharedPreferences sharedPref;

  checkUserExsistance(
      String phoneNo, onSuccess(isExsist), onError(error)) async {
    try {
      await http
          .get(Uri.parse(BASE_URL + "users/check_user/$phoneNo/"))
          .then((res) {
        if (res.statusCode == 200) {
          onSuccess(jsonDecode(res.body)['exists']);
        } else {
          throw jsonDecode(res.body)['ERR'];
        }
      }).catchError((err) {
        throw err.toString();
      });
    } catch (e) {
      onError(e);
    }
  }

  checkUserSession(String phoneNo, onSuccess(isExsist), onError(error)) async {
    sharedPref = await SharedPreferences.getInstance();
    try {
      await http.get(Uri.parse(BASE_URL + 'users/check_session/$phoneNo/'),
          headers: {
            'Authorization': sharedPref.getString('jwt').toString()
          }).then((res) {
        Map data = jsonDecode(res.body);
        if (res.statusCode == 200) {
          onSuccess(data['exists']);
        }
      }).catchError((err) {
        throw err.toString();
      });
    } catch (e) {
      onError(e);
    }
  }

  createAccountAtBackend(
      {firstName,
      lastName,
      phone,
      email,
      password,
      pincode,
      address,
      ref_code,
      required onSuccess(data),
      required onError(error)}) async {
    sharedPref = await SharedPreferences.getInstance();
    Map formData = new Map();

    formData['name'] = firstName + " " + lastName;
    formData['email'] = email;
    formData['phone'] = phone;
    formData['pincode'] = pincode;
    formData['password'] = password;
    formData['is_subdealer'] = 'false';
    formData['address'] = address;
    formData['ref_code'] = ref_code == null ? "" : ref_code;

    try {
      await http
          .post(
        Uri.parse(BASE_URL + "users/"),
        body: formData,
      )
          .then((res) async {
        Map data = jsonDecode(res.body);
        if (res.statusCode == 200) {
          sharedPref.setString('jwt', data['jwt']);
          sharedPref.setString('name', data['user']['name']);
          onSuccess(data);
        } else {
          throw data['ERR'];
        }
      }).catchError((err) {
        throw err.toString();
      });
    } catch (err) {
      onError(err);
    }
  }

  loginAtBackend(
      String phone, String password, onSuccess(res), onError(err)) async {
    sharedPref = await SharedPreferences.getInstance();
    try {
      await http.post(Uri.parse(BASE_URL + 'users/login/'),
          body: {'phone': phone, 'password': password}).then((res) {
        if (res.statusCode == 200) {
          Map data = jsonDecode(res.body);
          sharedPref.setString('jwt', data['jwt']);
          sharedPref.setString('name', data['user']['name']);
          onSuccess(res);
        } else {
          throw jsonDecode(res.body)['ERR'];
        }
      }).catchError((onError) {
        throw onError.toString();
      });
    } catch (e) {
      onError(e.toString());
    }
  }

  logoutAtBackend(onSuccess(data), onError(err)) async {
    sharedPref = await SharedPreferences.getInstance();

    try {
      await http.get(Uri.parse(BASE_URL + 'users/logout/'), headers: {
        'Authorization': sharedPref.getString('jwt').toString()
      }).then((res) {
        if (res.statusCode == 200) {
          sharedPref.clear();
          onSuccess(jsonDecode(res.body)['INFO']);
        } else {
          throw jsonDecode(res.body)['ERR'];
        }
      }).catchError((err) {
        throw err.toString();
      });
    } catch (e) {
      onError(e);
    }
  }

  updateProfileAtBackend(Map body, onSuccess(data), onError(err)) async {
    sharedPref = await SharedPreferences.getInstance();
    try {
      await http.put(
        Uri.parse(BASE_URL + 'users/${sharedPref.getString('jwt').toString().split('.')[0]}/'),
        headers: {
          "Content-Type": "application/x-www-form-urlencoded",
          'Authorization': sharedPref.getString("jwt").toString()  
        },
        body: body,
        encoding: Encoding.getByName('utf-8')
      ).then((res) {
        if (res.statusCode == 200) {
          sharedPref.setString('name', jsonDecode(res.body)['name']);
          onSuccess(res.body);
        } else {
          throw jsonDecode(res.body)['ERR'];
        }
      }).catchError((onError) {
        throw onError;
      });
    } catch (e) {
      onError(e);
    }  

  }

  getProfileData(onSuccess(data), onError(err)) async {
    sharedPref = await SharedPreferences.getInstance();
    
    try {
      await http.get(
      Uri.parse(BASE_URL + 'users/${sharedPref.get('jwt').toString().split('.')[0]}/'),
      headers: {
        'Authorization': sharedPref.getString("jwt").toString()
      }
    ).then((res) {
      if (res.statusCode == 200) {
        onSuccess(jsonDecode(res.body));
      } else {
        throw jsonDecode(res.body)['ERR'];
      }
    }).catchError((onError){throw onError.toString();});
    } catch (e) {
      onError(e);
    }
  }
}
